tests/cnf/core/network/metallb/tests/bgp-connect-time-tests.go

```diff
@@ -0,0 +1,224 @@
+				return connectTime
+			}, 60*time.Second, 5*time.Second).Should(ContainSubstring("ConnectRetryTimer: 10"),
+				"Fail to find received BGP connect time")
+
+			By("Reset the BGP session ")
+			err := frr.ResetBGPConnection(frrPod)
+			Expect(err).ToNot(HaveOccurred(), "Failed to reset BGP connection")
+
+			By("Verify that BGP session is established and up in less then 10 seconds")
+			verifyMetalLbBGPConnectTime(frrPod, removePrefixFromIPList(ipv4NodeAddrList))
+		})
+
+	It("Update the timer to less then the default on an existing BGP connection",
+		reportxml.ID("74417"), func() {
+
+			By("Creating BGP Peers")
+			createBGPPeerAndVerifyIfItsReady(ipv4metalLbIPList[0], "", 64500, false, frrk8sPods)
+
+			By("Validate BGP Peers with the default retry connect timer")
+			Eventually(func() string {
+				// Get the routes
+				connectTime, err := frr.VerifyBGPConnectTime(frrk8sPods, ipv4metalLbIPList[0], 120)
+				Expect(err).ToNot(HaveOccurred(), "Failed to verify BGP connect time")
+
+				// Return the routes to be checked
+				return connectTime
+			}, 60*time.Second, 5*time.Second).Should(ContainSubstring("ConnectRetryTimer: 120"),
+				"Fail to find received BGP connect time")
+
+			By("Update the BGP Peers connect timer to 10 seconds")
+			bgpPeer, err := metallb.PullBGPPeer(APIClient, "testpeer", NetConfig.MlbOperatorNamespace)
+			Expect(err).ToNot(HaveOccurred(), "Failed to find bgp peer")
+
+			_, err = bgpPeer.WithConnectTime(metav1.Duration{Duration: 10 * time.Second}).Update(true)
+			Expect(err).ToNot(HaveOccurred(), "Failed to update the bgp peer with a 10s connect timer")
+
+			By("Validate BGP Peers with the default retry connect timer")
+			Eventually(func() string {
+				// Get the routes
+				connectTime, err := frr.VerifyBGPConnectTime(frrk8sPods, ipv4metalLbIPList[0], 10)
+				Expect(err).ToNot(HaveOccurred(), "Failed to verify BGP connect time")
+
+				// Return the routes to be checked
+				return connectTime
+			}, 60*time.Second, 5*time.Second).Should(ContainSubstring("ConnectRetryTimer: 10"),
+				"Fail to find received BGP connect time")
+		})
+})
+
+func createAndDeployFRRPod() *pod.Builder {
+	var err error
+
+	By("Creating External NAD")
+	createExternalNad()
+
+	By("Creating static ip annotation")
+
+	staticIPAnnotation := pod.StaticIPAnnotation(
+		externalNad.Definition.Name, []string{fmt.Sprintf("%s/%s", ipv4metalLbIPList[0], "24")})
+
+	By("Creating MetalLb configMap")
+
+	masterConfigMap := createConfigMap(tsparams.LocalBGPASN, ipv4NodeAddrList, false, false)
+
+	By("Creating FRR Pod")
+
+	frrPod := createFrrPod(
+		masterNodeList[0].Object.Name, masterConfigMap.Definition.Name, []string{}, staticIPAnnotation)
+
+	time.Sleep(20 * time.Second)
+	Eventually(func() error {
+		return err
+	}, time.Minute, tsparams.DefaultRetryInterval).ShouldNot(HaveOccurred(),
+		"Failed to collect metrics from speaker pods")
+
+	return frrPod
+}
+
+func verifyMetalLbBGPConnectTime(frrPod *pod.Builder, peerAddrList []string) {
```

Comment:
the connect time should be a param here (not hard coded to 10s)

Reply:
THe timer there is how long for Eventual to wait. I set it to 10 seconds.

Reply:
ah right, so where does the function "verifies connect time" as the name suggests?